src/Hl7.Fhir.ElementModel/ElementNode.cs

```diff
@@ -94,6 +94,37 @@ public ElementNode Add(IStructureDefinitionSummaryProvider provider, string name
             return Add(provider, child);
         }
 
+        public ElementNode Insert (IStructureDefinitionSummaryProvider provider, ElementNode child, int index, string name = null)
```

Comment:
To avoid confusion I would keep only this overload, and without the `name` parameter (caller should set the correct name in `child`).

Reply:
I agree